### PR TITLE
bugfix/string-agg-removal

### PR DESCRIPTION
**Pull request by fivetran-joemarkiewicz**
## PR Overview
**This PR will address the following Issue/Feature:** Issue #118 

**This PR will result in the following new package version:** `v1.7.0`

This will be a breaking change as we are removing fields from an end model and could cause downstream errors with their removal.

**Please provide the finalized CHANGELOG entry which details the relevant changes included in this PR:**

### 🚨 Breaking Changes 🚨: Bug Fixes
- The following fields have been deprecated (removed) as these fields proved to be problematic across warehouses due to the end size of the fields.
  - `errors_since_last_completed_sync`
  - `warnings_since_last_completed_sync`
> Note: If you found these fields to be relevant, you may still reference the error/warning messages from within the underlying `log` table.
- The `fivetran_platform_using_sync_alert_messages` variable has been removed as it is no longer necessary due to the above changes.

### Feature Updates
- The following fields have been added to display the number of error/warning messages sync last completed sync. These fields are intended to substitute the information from deprecated fields listed above.
  - `number_errors_since_last_completed_sync`
  - `number_warnings_since_last_completed_sync`

### Basic Validation
Please acknowledge that you have successfully performed the following commands locally:
- [X] dbt run –full-refresh && dbt test
- [X] dbt run (if incremental models are present) && dbt test

Before marking this PR as "ready for review" the following have been applied:
- [X] The appropriate issue has been linked, tagged, and properly assigned
- [X] All necessary documentation and version upgrades have been applied
- [X] docs were regenerated (unless this PR does not include any code or yml updates)
- [X] BuildKite integration tests are passing
- [x] Detailed validation steps have been provided below

### Detailed Validation
Please share any and all of your validation steps:

Please see the corresponding Height ticket for a link to the Hex document that includes the Hex validations notebook.

### If you had to summarize this PR in an emoji, which would it be?
 
🔢

### CODE REVIEW

File: models/fivetran_platform__connector_status.sql
Comment (fivetran-reneeli) [inline models/fivetran_platform__connector_status.sql:202]: Will `message_data` always be different across each log record for each connector_id? What happens if there's 2 of the same message_data -- is it intended that they will only be counted once as the same error or warning?

File: models/fivetran_platform.yml
Comment (fivetran-reneeli) [inline models/fivetran_platform.yml:60]:
```suggestion
        description: Numeric metric indicating the number of warning messages raised since the last sync completion.
```

File: models/fivetran_platform.yml
Comment (fivetran-reneeli) [inline models/fivetran_platform.yml:57]:
```suggestion
      - name: number_errors_since_last_completed_sync
        description: Numeric metric indicating the number of error messages raised since the last sync completion.
```

File: README.md
Comment (fivetran-reneeli) [inline README.md:72]:
```suggestion
> Note that although the source connector is now "Fivetran Platform", the package retains the old name of "fivetran_log".
```

**Review by fivetran-reneeli — CHANGES_REQUESTED** [review]
Reviewed with a few suggestions/ question!

File: models/fivetran_platform__connector_status.sql
Comment (fivetran-joemarkiewicz) [reply models/fivetran_platform__connector_status.sql:202]: Good question, in this case we are just using the `message_data` field to count the warning/error entries. The distinct case is not being applied to the message_data, but instead the event_type. So if two message_data of the exact same verbiage are found in the warning event_type, then both will still be counted as a warning since last sync.

**Comment by fivetran-joemarkiewicz** [comment]
Thanks @fivetran-reneeli for the quick and thoughtful review! I just addressed your comments and committed your suggestions. Thanks for catching my spelling mistakes 🤦. I also addressed the open question you had and will hold off regenerating the docs until approval.

**Comment by fivetran-reneeli** [comment]
Thanks @fivetran-joemarkiewicz , just responded above!

File: models/fivetran_platform__connector_status.sql
Comment (fivetran-joemarkiewicz) [reply models/fivetran_platform__connector_status.sql:202]: Ah yes you're right! I was trying to understand why we used distinct previously, but that was to ensure we do not aggregate the same string over and over again. However, like you mentioned we do in fact want to still count those as error warnings with this new field. I will update to remove the distinct clause in the count statement.

Comment (fivetran-joemarkiewicz) [reply models/fivetran_platform__connector_status.sql]: Thanks for catching this and continuing to dive into this query! Just updated.

**Comment by fivetran-joemarkiewicz** [comment]
@fivetran-reneeli thanks again! I agree with your comment and just applied the changes and regenerated the docs. Let me know if you have any other comments.

File: models/fivetran_platform__connector_status.sql
Comment (fivetran-reneeli) [reply models/fivetran_platform__connector_status.sql]: Gotcha, thanks for digging into why we had it in place before! Looks good!

**Review by fivetran-reneeli — APPROVED** [review]
lgtm

**Review by fivetran-jamie — APPROVED** [review]

**Pull request merged**
